### PR TITLE
(towards #2543) Simplify Introduction documentation 

### PR DESCRIPTION
**Pull request by sergisiso**
- Added "warnings as errors" in the shpinx CI job

- Renamed "nemo" to "generic" in tutorials and lib folders.

- I removed lots of documentation that seem unneeded: two files where not used, some explanation e.g. how to use venvs is better with a link to an authoritative the source, how to install each dependency separately is probably not needed....

- Merged tutorials and examples sections and rewrote tutorials part

- I made termcolour and graphviz standard dependencies

- I started using sphinx tabs for alternative options, see image:
image

### CODE REVIEW

**Comment by sergisiso** [comment]
@arporter @hiker This is ready for review. A bit unfortunately the linkcheck does not pass because the last commit updated a link to https://github.com/stfc/PSyclone/tree/master/tutorial/practicals/generic which won't exist until itself is merged, but you can check in the previous commit that the tests pass with the warning as errors flags.

File: doc/user_guide/getting_going.rst [hunk 1]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
 
-pyparsing
-^^^^^^^^^
 
-PSyclone requires ``pyparsing``, a library designed to allow parsers to
-be built in Python. PSyclone uses ``pyparsing`` to parse Fortran regular
-expressions as ``fparser`` does not fully parse these (see
-`here <https://github.com/pyparsing>`__ for more information).
+    For more information about using ``pip`` or encapsulating the installation
+    in its own ``virtual environment`` we recomment reading the
```
Comment (arporter) [inline doc/user_guide/getting_going.rst:76]: "recommend"

**Review by arporter — CHANGES_REQUESTED** [review]
Nice job Sergi. Thanks very much for all your work on improving the documentation.
Mostly small things. The only concern I have is that the funky new 'tab' option doesn't render very well in latexpdf but perhaps we can tweak it (or just live with it)?
docs build fine for me (and this is now tested in the CI which is great).

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -66,11 +66,18 @@ jobs:
     - run: sudo apt-get install -y graphviz doxygen
     - run: python -m pip install --upgrade pip
     - run: pip install .[doc]
+    # First we need a first build and launching a server, otherwise the
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply .github/workflows/python-package.yml:69]: Done

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -66,11 +66,18 @@ jobs:
...
+    # intershpinx links would be broken
     - run: cd doc/user_guide; make html
     - run: cd doc/developer_guide; make html
     - run: cd doc/reference_guide; make html
     - run: cd doc/psyad/user_guide; make html
     - run: cd doc; python -m http.server &
+    # Now we can check for warnings and broken links
+    - run: cd doc/user_guide; make html SPHINXOPTS="-W --keep-going"
+    - run: cd doc/developer_guide; make html SPHINXOPTS="-W --keep-going"
+    # - run: cd doc/reference_guide; make html
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply .github/workflows/python-package.yml:79]: Done

File: doc/user_guide/getting_going.rst [hunk 4]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
-Location and structure of PSyclone installation
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Location of installed Pyclone scripts, modules and other accompanying
-resources is similar to other Python packages:
-
-* The ``psyclone`` :ref:`script <psyclone_command>` is located
-  in ``<python-base-prefix>/bin`` directory (depending on your Linux
-  distribution, you may need to add this location to your ``$PATH``).
+please see :ref:`Installation section in the Developer Guide <dev-installation>`.
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:50]: Done

File: doc/user_guide/getting_going.rst (hunk 1 above)
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:76]: Done

File: doc/user_guide/getting_going.rst [hunk 5]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
+    .. code-block:: bash
 
-   > pip install pyparsing
+      spack install psyclone
 
-Should you wish to, uninstalling is simply performed by doing:
-::
 
-   > pip uninstall pyparsing
+    For more information about how to use Spack we recommend readign the
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:88]: Fixed

File: doc/user_guide/getting_going.rst [hunk 6]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
-is installed, this command will complete successfully. If ``sympy`` is
-installed you can check its version by typing
-``sympy.__version__`` after successfully importing it.
+Installation location
+^^^^^^^^^^^^^^^^^^^^^
 
-If ``sympy`` is not installed on your system then it may be installed
-from the Python Package Index using ``pip``:
-::
+PSyclone installation location will vary depending on the specific installation
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:109]: Done

File: doc/user_guide/getting_going.rst [hunk 7]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
+method and options used. The ``pyclone`` command will typically already be
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:110]: Done

File: doc/user_guide/getting_going.rst [hunk 8]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
-
-Installation (and uninstallation) of this package can be done via
-``pip`` in exactly the same way as for ``graphviz``, as described above.
 
 .. _getting-going-configuration:
 
 Configuration
 -------------
 
 Various aspects of PSyclone are configured through a configuration
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:128]: Done

File: doc/user_guide/getting_going.rst [hunk 9]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
 .. warning::
 
-   If PSyclone is installed to a non-standard location (e.g. by specifying
-   the ``--install-option="--prefix=...`` option to ``pip install``) then
-   PSyclone will not be able to find the configuration file at execution
-   time. There are two solutions to this: 1. copy the configuration file to
-   a location where PSyclone will find it (see :ref:`configuration`) or
-   2. set the ``PSYCLONE_CONFIG`` environment variable to the full-path to
-   the configuration file, e.g.::
+   If PSyclone is installed in 'editable' mode (``-e`` flag to ``pip``), ``pip``
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:134]: Deleted

File: doc/user_guide/getting_going.rst [hunk 10]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
+   or in a non-standard location, then PSyclone will not be able to find the
+   default configuration file. There are two solutions to this:
 
-   > export PSYCLONE_CONFIG=/some/path/PSyclone/config/psyclone.cfg
+   1. copy a configuration file to the location specified above.
 
-.. warning::
+   2. set the ``PSYCLONE_CONFIG`` environment variable to the full-path to
+   the configuration file, e.g.:
+
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:142]: Added mention to it in parenthesis

File: doc/user_guide/getting_going.rst [hunk 11]
```diff
@@ -41,426 +41,121 @@ Getting Going
...
-
-* The ``psyclone`` :ref:`script <psyclone_command>` is located
-  in ``<python-base-prefix>/bin`` directory (depending on your Linux
-  distribution, you may need to add this location to your ``$PATH``).
+please see :ref:`Installation section in the Developer Guide <dev-installation>`.
 
-* The PSyclone Python modules are located in
-  ``<python-base-prefix>/lib/pythonX.Y/site-packages`` directory (where
-  ``X.Y`` is the version of Python that you are using).
+.. tabs::
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:52]: I see, what do you think if they are uppercase and with colon?

web:
image
pdf:
image

File: doc/user_guide/getting_going.rst [hunk 12]
```diff
@@ -513,12 +208,12 @@ command. To list the available options run: `psyclone -h`, it should output::
...
-There is more detailed information about each flag in the :ref:`psyclone_command` section,
+There is more detailed information about each flag in :ref:`psyclone_command` section,
 but the main parameters are the input source file that we aim to transform, and a transformation
-recipe that is provided with the `-s` flag.
-In addition to these, note that ``psyclone`` can be used in two distinct modes:
-the code-transformation mode (when no `-api`/`--psykal-dsl` flags is provided) or the
-PSyKAl DSL mode (when a `-api`/`--psykal-dsl` flag is provided). The following sections provide
+recipe that is provided with the ``-s`` flag.
+In addition to these, note that psyclone can be used in two distinct modes:
+the code-transformation mode (when no ``-api``/``--psykal-dsl`` flags is provided) or the
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/getting_going.rst:215]: Done

File: doc/user_guide/tutorials_and_examples.rst [hunk 13]
```diff
@@ -35,16 +35,47 @@
...
+
+PSyclone provides a hands-on tutorial. The easiest way to follow it is reading
+the `Readme files in github <https://github.com/stfc/PSyclone/tree/master/tutorial/practicals>`_.
+The tutorial is divided in two sections, a first section that introduces
+PSyclone and how to
+`use it to transform generic Fortran code  <https://github.com/stfc/PSyclone/tree/master/tutorial/practicals/generic>`_
+(this is the recommended starting point for everybody).
+And a second sections about
+`the LFRic DSL <https://github.com/stfc/PSyclone/tree/master/tutorial/practicals/LFRic>`_
+(this is only recommended for people interested in PSKAL DSLs and LFRic in particular).
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/tutorials_and_examples.rst:55]: Fixed

File: doc/user_guide/tutorials_and_examples.rst [hunk 14]
```diff
@@ -35,16 +35,47 @@
...
+
+To do the proposed hands-on you will need a linux shell with Python installed and to
+download the hand-on directory with:
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/tutorials_and_examples.rst:58]: Fixed

File: doc/user_guide/tutorials_and_examples.rst [hunk 15]
```diff
@@ -35,16 +35,47 @@
...
+
+Tutorial and Examples
+=====================
+
+Tutorial
+--------
+
+PSyclone provides a hands-on tutorial. The easiest way to follow it is reading
+the `Readme files in github <https://github.com/stfc/PSyclone/tree/master/tutorial/practicals>`_.
+The tutorial is divided in two sections, a first section that introduces
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/tutorials_and_examples.rst:49]: Fixed

File: doc/user_guide/tutorials_and_examples.rst [hunk 16]
```diff
@@ -749,21 +800,3 @@ For more details see the ``examples/nemo/README.md`` file.
 Note that these scripts are here to support the ongoing development of the
 NEMO API in PSyclone. They are *not* intended as 'turn-key' solutions but
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply doc/user_guide/tutorials_and_examples.rst:801]: Done

File: src/psyclone/tests/conftest.py [hunk 17]
```diff
@@ -81,10 +81,11 @@ def have_graphviz():
     ''' Whether or not the system has graphviz installed. Note that this
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/tests/conftest.py:81]: Updated

File: tutorial/practicals/generic/Makefile [hunk 18]
```diff
@@ -35,6 +35,6 @@
 
 # Top-level Makefile to test all of the NEMO hands-on tutorials.
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply tutorial/practicals/generic/Makefile:36]: Done

**Comment by sergisiso** [comment]
@arporter This is ready for another review

File: doc/user_guide/getting_going.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (arporter) [reply doc/user_guide/getting_going.rst:53]: Oh yes, that's better.

File: doc/user_guide/tutorials_and_examples.rst [hunk 19]
```diff
@@ -749,21 +800,3 @@ For more details see the ``examples/nemo/README.md`` file.
...
 as a starting point.
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/user_guide/tutorials_and_examples.rst]: Great, thanks.

**Review by arporter — APPROVED** [review]
All looks good now. Thanks very much.
Will proceed to merge.

**Pull request merged**